### ISSUE TITLE
Doesn't handle $$ in filenames correctly

### ISSUE THREAD

**Issue by krainboltgreene** [AgentME/browserify-hmr#29]
Ran into this error:

```
Uncaught TypeError: Cannot set property 'module' of undefined
```

Turns out there's a bug with the filenames?

<img width="503" alt="screen shot 2016-04-28 at 1 57 23 am" src="https://cloud.githubusercontent.com/assets/334809/14880625/c3bd247c-0ce4-11e6-9fe8-0ec300a66e81.png">
<img width="228" alt="screen shot 2016-04-28 at 1 57 28 am" src="https://cloud.githubusercontent.com/assets/334809/14880626/c3c1674e-0ce4-11e6-8ee9-822bdeaf0976.png">
<img width="613" alt="screen shot 2016-04-28 at 1 57 32 am" src="https://cloud.githubusercontent.com/assets/334809/14880624/c3bc55e2-0ce4-11e6-8fb2-45a6b6011844.png">

Have you considered checksums?

**Comment by krainboltgreene** [comment]
Turns out this was node-fs-extra's fault, but since it depends on it I suggest keeping this open until they fix and have an update pushed.

I also still suggest using checksums for the property names, since filenames can be really crazy.

**Comment by AgentME** [comment]
Nah, there was a real issue in browserify-hmr here. Turns out if you do `someString.replace(foo, bar)`, then if `bar` is a string with `$` characters, they get treated specially. The fix was to change `bar` to be a function that returns a string, which .replace doesn't do anything special with.

Fixed in browserify-hmr 0.3.2.
